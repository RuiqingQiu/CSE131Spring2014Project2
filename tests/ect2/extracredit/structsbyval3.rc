const int c = 15;


structdef S { int [c] a; };
structdef S2 { S a; S b; S c; };

function : S foo (S x, S y, S z){

   //S2 r;

   int i = 0;
   while (i < c){
      y.a[i] = (z.a[i] = x.a[i]  + c) + c;
      cout << x.a[i] << "." << y.a[i] << "." << z.a[i] << endl;
   i++;}
    
   return x;
   //r.a = x; r.b = y; r.c = z;
   //return r;
}

function : int main(){  
   S x, y, z;
   
   int i = 0; 
   
   while (i < c){
      x.a[i] = (y.a[i] = (z.a[i] = i) + 1) + 1;
      cout << x.a[i] << "." << y.a[i] << "." << z.a[i] << endl;
   i++;}
   
   //S2 a;
   S a = foo(x,y,z);
   
   while (i < c){
      x.a[i] = (y.a[i] = (z.a[i] = i*3) + 5) + 7;
   i++;}
   
   //S2 b = foo(x,y,z);
   
   i = 0;
   while (i < c){
     // cout << a.a.a[i] << "." << a.b.a[i] << "." <<  a.c.a[i++] << endl;
     cout << a.a[i++] << endl;
   }

   

   return 0;
}
