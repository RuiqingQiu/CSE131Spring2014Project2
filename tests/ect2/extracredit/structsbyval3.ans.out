2.1.0
3.2.1
4.3.2
5.4.3
6.5.4
7.6.5
8.7.6
9.8.7
10.9.8
11.10.9
12.11.10
13.12.11
14.13.12
15.14.13
16.15.14
2.32.17
3.33.18
4.34.19
5.35.20
6.36.21
7.37.22
8.38.23
9.39.24
10.40.25
11.41.26
12.42.27
13.43.28
14.44.29
15.45.30
16.46.31
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
